File: movies-recommender-service/src/main/kotlin/org/miejski/movies/recommender/infrastructure/dbstate/assertions/Similarity_1_Assertion.kt
package org.miejski.movies.recommender.infrastructure.dbstate.assertions

import org.miejski.movies.recommender.infrastructure.dbstate.Neo4jStateAssertion
import org.neo4j.ogm.session.Session
import java.util.*

class Similarity_1_Assertion(val session: Session) : Neo4jStateAssertion {
    override fun name(): String {
        return "Similarity_1_Assertion"
    }

    override fun queryToExecute(): String {
        return "start_state/similarity_pearson.cypher"
    }

    override fun isOK(): Boolean {
        return session.query("Match (p:Person)-[s:Similarity]-(p2:Person) where exists(s.similarity) return s limit 1", HashMap<String, Object>()).count() > 0
    }
}

File: deployment/scripts/neo4j_state/assertions/similarity_1_assertion.py
class Similarity1Assertion:

    def is_ok(self, Neo4jCypherExecutor):
        result = Neo4jCypherExecutor.invoke(self.__check_query())
        return len(result) > 0

    def query_to_execute(self):
        return "similarity_1.cypher"

    def arguments(self):
        return {}

    def __check_query(self):
        return "MATCH (p:Person)-[s:Similarity]->(p2:Person) return s limit 1"
